low-level/framework/utils/consulstore.py

```diff
@@ -100,9 +100,14 @@ def _consul_get(self, key, **kwargs):
                 time.sleep(WAIT_BEFORE_RETRY)
 
             except Exception as gerr:
-                logger.warn("Error[{0}] while reading data from consul {1}" \
-                    .format(gerr, key))
-                break
+                if 'no cluster leader' in gerr.lower():
```

Comment:
Can we add similar logic for other methods here too? e.g. put, delete, etc. even __init__ too if possible? Please check all possible places and change accordingly.

Reply:
Done

Reply:
ok